### PR TITLE
Save shopping list between sessions and display declined request

### PR DESCRIPTION
**Pull request by AsafGartner**

### CODE REVIEW

File: app/src/main/java/com/iic/shopingo/ui/request_flow/activities/RequestStateActivity.java [hunk 1]
```diff
@@ -40,19 +50,37 @@
...
-    ButterKnife.inject(this);
-    String explanationString = getString(EXPLANATION_STRINGS[request.getStatus().ordinal()]);
-    statusExplanation.setText(String.format(explanationString, request.getShopper().getFirstName()));
+
+  protected void onStart() {
+    super.onStart();
+    if (firstStart && getIntent().hasExtra(EXTRAS_REQUEST_KEY)) {
+      setRequest((OutgoingRequest)getIntent().getParcelableExtra(EXTRAS_REQUEST_KEY));
+    } else {
+      setRequest(new OutgoingRequestStorage(PreferenceManager.getDefaultSharedPreferences(this)).load());
```
Comment (assafgelber) [inline app/src/main/java/com/iic/shopingo/ui/request_flow/activities/RequestStateActivity.java]: Wouldn't it make more sense to load the request in the home activity and pass it in through the intent? That way this activity wouldn't care how it's loaded..

**Comment by assafgelber** [comment]
:cool::cool::cool:

File: app/src/main/java/com/iic/shopingo/ui/request_flow/activities/RequestStateActivity.java (hunk 1 above)
Comment (AsafGartner) [inline app/src/main/java/com/iic/shopingo/ui/request_flow/activities/RequestStateActivity.java]: I think it really depends on how we end up doing the notifications. Dunno at the moment.

Comment (assafgelber) [inline app/src/main/java/com/iic/shopingo/ui/request_flow/activities/RequestStateActivity.java]: k

Comment (ifeins) [inline app/src/main/java/com/iic/shopingo/ui/request_flow/activities/RequestStateActivity.java]: Let's just pass it on the intent.
I'm going to change my GcmIntentService to handled accepted/declined requests by creating an OutgoingRequest and  setting as an extra for the RequestStateActivity.

File: app/src/main/res/layout/activity_request_state.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+</FrameLayout>
```
Comment (ifeins) [inline app/src/main/res/layout/activity_request_state.xml]: How come this layout is needed now, when before it wasn't?

**Comment by ifeins** [comment]
We need to make sure it plays nicely with notifications, because currently when a notification is being handled the data is loaded from the server, now potentially it can be loaded from the local storage.

**Pull request merged**